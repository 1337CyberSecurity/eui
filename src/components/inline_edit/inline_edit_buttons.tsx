/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent } from 'react';
import { CommonProps } from '../common';
import classNames from 'classnames';
import { EuiButtonIcon } from '../button';
import { EuiFormRow } from '../form';
import { EuiTextProps } from '../text';
import { EuiTitleSize } from '../title';
import { EuiFlexItem } from '../flex';
import { useEuiI18n } from '../i18n';
import { getInlineEditIconButtonSettings } from './inline_edit_utils';

export type EuiInlineEditButtonsProps = CommonProps & {
  size?: EuiTextProps['size'] | EuiTitleSize;
  saveButtonAriaLabel?: string;
  cancelButtonAriaLabel?: string;
  saveFunction: Function;
  cancelFunction: Function;
};

export const EuiInlineEditButtons: FunctionComponent<EuiInlineEditButtonsProps> = ({
  className,
  size = 'm',
  saveButtonAriaLabel,
  cancelButtonAriaLabel,
  saveFunction,
  cancelFunction,
}) => {
  const classes = classNames('euiInlineEditButtons', className);

  const defaultSaveButtonAriaLabel = useEuiI18n(
    'euiInlineEditButtons.saveButtonAriaLabel',
    'Save edit'
  );

  const defaultCancelButtonAriaLabel = useEuiI18n(
    'euiInlineEditButtons.cancelButtonAriaLabel',
    'Cancel edit'
  );

  const buttonSettings = getInlineEditIconButtonSettings(size);

  return (
    <>
      <EuiFlexItem grow={false} className={classes}>
        <EuiFormRow>
          <EuiButtonIcon
            iconType="check"
            aria-label={saveButtonAriaLabel || defaultSaveButtonAriaLabel}
            onClick={() => saveFunction()}
            color="success"
            display="base"
            size={buttonSettings.compressed ? 's' : 'm'}
            iconSize={buttonSettings.iconSize}
          />
        </EuiFormRow>
      </EuiFlexItem>

      <EuiFlexItem grow={false}>
        <EuiFormRow>
          <EuiButtonIcon
            iconType="cross"
            aria-label={cancelButtonAriaLabel || defaultCancelButtonAriaLabel}
            onClick={() => cancelFunction()}
            color="danger"
            display="base"
            size={buttonSettings.compressed ? 's' : 'm'}
            iconSize={buttonSettings.iconSize}
          />
        </EuiFormRow>
      </EuiFlexItem>
    </>
  );
};
