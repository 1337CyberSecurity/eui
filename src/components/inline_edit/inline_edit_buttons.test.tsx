/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { EuiInlineEditButtons } from './inline_edit_buttons';

describe('EuiInlineEditButtons', () => {
  describe('props', () => {
    test('renders InlineEdit buttons', () => {
      const onSave = jest.fn();
      const onCancel = jest.fn();

      const { container } = render(
        <EuiInlineEditButtons
          saveFunction={onSave}
          cancelFunction={onCancel}
          {...requiredProps}
        />
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
