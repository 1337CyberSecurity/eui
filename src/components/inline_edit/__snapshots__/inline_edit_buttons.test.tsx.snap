// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiInlineEditButtons props renders InlineEdit buttons 1`] = `
<div
  class="euiFlexItem emotion-euiFlexItem-growZero"
>
  <div
    class="euiFormRow"
    id="generated-id-row"
  >
    <div
      class="euiFormRow__fieldWrapper"
    >
      <button
        aria-label="Save edit"
        class="euiButtonIcon euiButtonIcon--medium emotion-euiButtonIcon-base-success"
        id="generated-id"
        type="button"
      >
        <span
          aria-hidden="true"
          class="euiButtonIcon__icon"
          color="inherit"
          data-euiicon-type="check"
        />
      </button>
    </div>
  </div>
</div>
`;
